homeassistant/components/esphome/assist_satellite.py

```diff
@@ -466,10 +469,18 @@ async def _wrap_audio_stream(self) -> AsyncIterable[bytes]:
             yield chunk
 
     def _stop_pipeline(self) -> None:
-        """Request pipeline to be stopped."""
+        """Request pipeline to be stopped by ending the audio stream and continue processing."""
         self._audio_queue.put_nowait(None)
         _LOGGER.debug("Requested pipeline stop")
 
+    def _abort_pipeline(self) -> None:
+        """Request pipeline to be aborted (no further processing)."""
+        _LOGGER.debug("Requested pipeline abort")
+        self._audio_queue.put_nowait(None)
+        if self._pipeline_task is not None:
+            self._pipeline_task.cancel()
+            self._pipeline_task = None
```

Comment:
The pipeline task should clean itself up, such that it is also cleaned up when it finishes. 
```suggestion
```